dashboard/utils.py

```diff
@@ -418,6 +418,19 @@ def get_pearson_exam_status(self):  # pylint: disable=too-many-return-statements
             )
             return ExamProfile.PROFILE_INVALID
 
+    def get_best_proctorate_exam_grade(self, course):
+
+        """
+        Returns the best exam grade
+
+        Args:
+            course (courses.models.Course): a course
+
+        Returns:
+            grades.models.ProctoredExamGrade: the best exam grade
+        """
+        return self.get_course_proctorate_exam_results(course).filter(passed=True).order_by('percentage_grade').last()
```

Comment:
Another thing that I discovered recently: We should only be considered `ProctoredExamGrade`s with an associated `ExamRun` that has a `date_grades_available` in the past. We might want to create a `ProctoredExamGradeQuerySet`, a bit like `FinalGradeQuerySet` is for `FinalGrade`. That could implement something like `with_available_grade` to check for that `ExamRun` date, and also some other helpful methods like `passed`.

Reply:
This is the queryset function `get_course_proctorate_exam_results`. It calls `for_user_course` class method that filters on `exam_run__date_grades_available__lte=now`.